src/main/java/com/aws/greengrass/deployment/errorcode/DeploymentErrorCodeUtils.java

```diff
@@ -180,38 +179,88 @@ private static void collectErrorCodesFromS3Exception(Set<DeploymentErrorCode> er
+
+    /**
+     * Assign error type to deployment request errors.
+     *
+     * @param deploymentType deployment type
+     * @return nucleus error if local deployment; cloud service error if cloud deployment
+     */
+    public static DeploymentErrorType getDeploymentRequestErrorType(Deployment.DeploymentType deploymentType) {
+        switch (deploymentType) {
+            // if a local deployment request is invalid, then it's a bug in CLI and mark it as Nucleus error
+            case LOCAL:
+                return DeploymentErrorType.NUCLEUS_ERROR;
+            // if cloud deployment, then mark it cloud service error
+            case SHADOW:
+            case IOT_JOBS:
+                return DeploymentErrorType.CLOUD_SERVICE_ERROR;
+            default:
+                return DeploymentErrorType.UNKNOWN_ERROR;
         }
     }
 
     /**
-     * Check whether a component is 1p.
+     * Check whether a service is 1p.
      *
-     * @param arn component arn from metadata
-     * @return true if it's an AWS component
-     * @throws PackageLoadingException unrecognized arn
+     * @param service        service to be checked
+     * @param componentStore a reference of componentStore
+     * @return AWS component error if account is AWS; user component error if a 12 digit account; a generic component
+     *         error type if anything wrong happened.
      */
-    public static boolean isAWSComponent(@NonNull String arn) throws PackageLoadingException {
-        if (arn.isEmpty()) {
-            throw new PackageLoadingException("Empty component arn is loaded");
+    public static DeploymentErrorType classifyComponentError(GreengrassService service, ComponentStore componentStore) {
+        // get service topic for name and version
+        Topics serviceTopics = service.getServiceConfig();
+        if (serviceTopics == null) {
+            logger.atWarn().log("Null service topic while classifying component error");
+            return DeploymentErrorType.COMPONENT_ERROR;
         }
 
-        final Matcher matcher = COMPONENT_ARN_PATTERN.matcher(arn);
-        if (matcher.matches()) {
-            String accountId = matcher.group(5);
-            return "aws".equals(accountId);
+        // load component arn from recipe metadata json on disk
+        String arnString;
+        try {
+            arnString = componentStore.getRecipeMetadata(ComponentIdentifier.fromServiceTopics(serviceTopics))
+                    .getComponentVersionArn();
+        } catch (PackageLoadingException e) {
+            logger.atWarn().log("Failed to load component metadata file from disk while classifying component error.");
+            return DeploymentErrorType.COMPONENT_ERROR;
         }
-        throw new PackageLoadingException("Component arn loaded is not valid");
-    }
 
+        // parse the arn to check if account id is AWS
+        try {
+            Arn arn = Arn.fromString(arnString);
+
+            // sanity check service == greengrass
```

Comment:
why bother with all this? Why not just check for the accountId? This seems unnecessary.

Reply:
True I also wasn't sure if they were necessary. Will remove them.